validation/linux_cgroups_blkio.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/opencontainers/runtime-tools/cgroups"
+	"github.com/opencontainers/runtime-tools/validation/util"
+)
+
+func main() {
+	var weight uint16 = 500
+	var leafWeight uint16 = 300
+	var major, minor int64 = 8, 0
+	var rate uint64 = 102400
+	g := util.GetDefaultGenerator()
+	g.SetLinuxCgroupsPath("/test")
+	g.SetLinuxResourcesBlockIOWeight(weight)
+	g.SetLinuxResourcesBlockIOLeafWeight(leafWeight)
+	g.AddLinuxResourcesBlockIOWeightDevice(major, minor, weight)
+	g.AddLinuxResourcesBlockIOLeafWeightDevice(major, minor, leafWeight)
+	g.AddLinuxResourcesBlockIOThrottleReadBpsDevice(major, minor, rate)
+	g.AddLinuxResourcesBlockIOThrottleWriteBpsDevice(major, minor, rate)
+	g.AddLinuxResourcesBlockIOThrottleReadIOPSDevice(major, minor, rate)
+	g.AddLinuxResourcesBlockIOThrottleWriteIOPSDevice(major, minor, rate)
+	err := util.RuntimeOutsideValidate(g, func(path string) error {
+		cg, err := cgroups.FindCgroup()
+		if err != nil {
+			return err
+		}
+		lbd, err := cg.GetBlockIOData(path)
```

Comment:
I think `path` should point to the bundle directory, which means we'll need something else for finding the cgroup path.  I think we want two tests for absolute `cgroupsPath`:

* Get the container process PID using `state`, and read in `/proc/{pid}/cgroup` to get the associated paths.  Compare those paths with `cgroupsPath` to test “[the runtime MUST take the path to be relative to the cgroups mount point][1]”, and raise `CgroupsAbsPathRelToMount` on mismatch.
* Then read the cgroup limits from whichever group you read from `/proc/{pid}/cgroup` (which may or may not be at the correct path), and compare the value with the configured value.  Failures here should probably be non-spec warnings, because [the spec has no RFC 2119 requirements for cgroup values][2].

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/config-linux.md#L181
[2]: https://github.com/opencontainers/runtime-spec/issues/746#issuecomment-297482403